docs/conandata_yml_format.md

```diff
@@ -0,0 +1,201 @@
+# conandata.yml
```

Comment:
Documenting this file is absolutely needed, but I would organize it with fewer and longer sections. This is just a suggestion, feel free to ignore it (close/resolve it).

---


# conandata.yml

<the intro 👍 >

In the context of CCI, this file is mandatory and consist in two main sections that we will explain in the next sections with more detail:

 * `sources`: a paragraph to explain the purpose 
 * `patches`: a paragraph to explain the purpose


## sources

(The same paragraph as above) This is the entry that contains all the items that are downloaded from internet and used by a recipe. This section contains one entry per each version and, each version should declare its own sources.

This is a basic example of a regular library, it should satisfy most of the use cases:

```yml
sources:
   "1.2.11":
       url: "..."
       sha256: "..."
   "1.2.12":
       url: "..."
       sha256: "..."
```

Every entry for a version consists in a dictionary with the `url` and the hashing algorithm of the artifact. `sha256` is required, but others like `sha1` or `md5` can be used as well.

**Mirrors.-** Sometimes it is useful to declare mirrors, use a list in the `url` field. Conan will try to download the artifacts from any of those mirrors.

```yml
sources:
   "1.2.11":
       url:
        - "url1"
        - "url2"
       sha256: "..."
```

## patches

Sometimes it is needed to apply some modifications to downloaded sources... blabalbla...

This section follows the same pattern as the `sources` above: one entry per version with a list of patches to apply. 

```yaml
patches:
   "1.2.11":
     - patch_file: ...
       base_path: ...
       url: ...
       sha256: ...
       patch_type:  ...
       patch_source: ...
       patch_description: ..
```

These fields have the following purpose (using _bold_ for those that are required):
 * `patch_file`: ...
 * `base_path`: ...
 * ...


## Some special cases

Sometimes there are some libraries that need to download several artifacts from internet, or the sources are different for each configuration,... here you can find how to declare these use cases:

### Several artifacts per version

Example: source code + license file

```yml
sources:
  8.0.0:
    - url: https://github.com/approvals/ApprovalTests.cpp/releases/download/v.8.0.0/ApprovalTests.v.8.0.0.hpp
      sha256: e16a97081f8582be951d95a9d53dc611f1f5a84e117a477029890d0b34ae99d6
    - url: "https://raw.githubusercontent.com/approvals/ApprovalTests.cpp/v.8.0.0/LICENSE"
      sha256: c71d239df91726fc519c6eb72d318ec65820627232b2f796219e87dcf35d0ab4
```

### Different sources per configuration

```yml
sources:
  "0066":
    "Macos":
      "apple-clang":
        "x86":
          - url: "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/MacIntel_OSX_AppleC_32bit/packages/cspice.tar.Z"
            sha256: "9a4b5f674ea76821c43aa9140829da4091de646ef3ce40fd5be1d09d7c37b6b3"
        "x86_64":
          - url: "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/MacIntel_OSX_AppleC_64bit/packages/cspice.tar.Z"
            sha256: "f5d48c4b0d558c5d71e8bf6fcdf135b0943210c1ff91f8191dfc447419a6b12e"
```